pkg/server/tunnel.go

```diff
@@ -144,5 +144,18 @@ func (t *Tunnel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			"connectionID", connection.connectID)
 	}
 
+	packet := &client.Packet{
+		Type: client.PacketType_CLOSE_REQ,
+		Payload: &client.Packet_CloseRequest{
+			CloseRequest: &client.CloseRequest{
+				ConnectID: connID,
+			},
+		},
+	}
+
+	if err = backend.Send(packet); err != nil {
+		klog.ErrorS(err, "error sending close request packet", "host", r.Host, "agentID", agentID, "connID", connID)
```

Comment:
Most of the time if you get here it's because the relevant backend is already closed. As such this should not be an ErrorS. V(4).InfoS at most.

Reply:
The destination may have its connection closed but sending a CLOSE_REQ to a backend should never fail unless the agent goes away (or the network connection is broken). If that's the case, then I think the ErrorS log makes sense?

Reply:
For us to get here, a break had to happen earlier, that suggests we either got a EOF or an earlier Send received an error and was logged with a ErrorS. I would rather just see that message as it pertains to the actual problem. This seems more likely to mislead than help. In the EOF case it suggests we may have a worse problem than we do. In the failed on data request, seeing a failure on an attempt to close may trick us into thinking it was going down anyway.

Reply:
Maybe I'm misunderstanding this but this ServeHTTP faces the apiserver so an EOF implies that the client (apiserver) chose to close the connection. This seems fairly common so we should get into this code path quite often. I guess if this code path was reached via failing to send data to the backend, an error logged here would be misleading...but that shouldn't happen too often right?

Reply:
Sounds right. However if the client closed the connection, then we can expect this send to fail. I'd rather not have alarming errors in the log for what is the normal case. Even in the case where there was an actual error, we are still the second failure and the first legitimate failure will have been logged as an error.

Reply:
Discussed offline. V(2) might be more preferable but I agree we probably don't need this to be `ErrorS`.

Reply:
@cheftako @Jefftree thank you for your kind comments, i will modify it to kog.V(2).InfoS